src/mpid/ch4/netmod/ofi/ofi_pre.h

```diff
@@ -28,7 +28,8 @@
 #define MPIDI_OFI_AM_HANDLER_ID_BITS   8
 #define MPIDI_OFI_AM_TYPE_BITS         8
 #define MPIDI_OFI_AM_HDR_SZ_BITS       8
-#define MPIDI_OFI_AM_DATA_SZ_BITS     48
+#define MPIDI_OFI_AM_PAYLOAD_SZ_BITS  24
```

Comment:
why 24 bits?

Reply:
I was thinking of fitting the payload size and the seq_no into the same 64 bit region. Seq_no takes 16 bits, so payload size only gets 24 bits. Since the payload size is only describing the data size on this OFI message, I think 24 bits should be sufficient.

Reply:
I see. PIPELINE and RDMA READ transfers aren't bound by this limit, right? This is just for single (ofi_hdr+am_hdr+payload) messages?

Reply:
@raffenet Yes. This is only for (ofi_hdr+am_hdr+payload) message. The zcopy message is not bound by this and its format is going to be (ofi_hdr + am_hdr + zcopy_hdr). Should the implementation of the zcopy protocol requires the total message size being transmitted, it would just add it into the zcopy_hdr. This is pretty much what the OFI_imt_info carries today apart from mem registration info.